cli/azd/pkg/tools/bash/bash.go

```diff
@@ -5,23 +5,24 @@ import (
 	"runtime"
 	"strings"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
```

Comment:
We keep walking into increasing a complicated package-cross-dependency which is slowly buying debt.

Instead of making `bash` package depend on `environment`, consider adding an `environment` type within the `bash` package and export it. Then, anyone calling `NewBashScript` should know how to convert `environment.Env` to `bash.Env` to consume `bash` package.

This way, the Environment defined within the `bash` has the right model for bash. Instead of making `environment.Environ()` method just for `bash package`

Reply:
Thanks for the reminder.  Now that the environment reload is available we can rollback this change and just keep a simple array of env vars without any issue since these scripts are created/invoked as needed which would still have all the updated env values.